src/lib/Behat/Helper/UtilityContext.php

```diff
@@ -265,7 +265,7 @@ public function waitUntilElementDisappears(string $cssSelector, int $timeout): v
     {
         try {
             $this->waitUntil($timeout, function () use ($cssSelector, $timeout) {
-                return !$this->isElementVisible($cssSelector, $timeout);
+                return !$this->isElementVisible($cssSelector, 1);
```

Comment:
Could you explain this change? :) Looks like hardcoded value.

Reply:
Sure. The problem is that `isElementVisible` also has a (blocking) waitUntil inside it - when `$timeout` is passed as an argument then the `isElementVisible` will wait the whole `$timeout` for the Element to become visible - making the outer loop not working as intended.

Example (current behaviour)

```
$timeout = 5;
$this->waitUntil($timeout, function () use ($cssSelector, $timeout) {
                return !$this->isElementVisible($cssSelector, $timeout);
}
```

The first time `isElementVisible` is invoked there are two possibilities:
- Element is visible - function returns true immediately
- Element is not visible - function returns false after 5 seconds

Only then the (outer) waitUntil is taken into account:
- Element is visible - isElementVisible is run again
- Element is not visible - function passes

This means that in both scenarios (Element is visible all the time and Element not visible at all) the function will take $timeout seconds to execute it. Turning it into a more complex sleep 😛 

Example after changes:

```
$timeout = 5;
$this->waitUntil($timeout, function () use ($cssSelector, $timeout) {
                return !$this->isElementVisible($cssSelector, 1);
}
```

The first time `isElementVisible` is invoked there are two possibilities:
- Element is visible - function returns true immediately
- Element is not visible - function returns false after 1 second

Only then the (outer) waitUntil is taken into account:
- Element is visible - isElementVisible is run again
- Element is not visible - function passes

Result:
in one scenario (Element is visible all the time) the function takes $timeout seconds, in the other (Element not visible at all) the function takes 1 second to execute.

The current behaviour becomes a problem when $timeout values are getting bigger, for example 20-30 seconds.